File: app/src/main/java/com/arash/altafi/room/java/userDao.java
package com.arash.altafi.room.java;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public interface userDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(userModel model);

    @Query("SELECT * FROM tbl_user")
    List<userModel> getAll();

    @Delete
    void delete(userModel model);

    @Update
    void update(userModel model);

}

File: app/src/main/java/com/arash/altafi/room/kotlin/user_dao.kt
package com.arash.altafi.room.kotlin

import androidx.room.*

@Dao
interface user_dao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(userModel: user_model)

    @Query("SELECT * FROM tbl_user_kotlin")
    fun getAll() : List<user_model>

    @Delete
    fun delete(userModel: user_model)

    @Update
    fun update(userModel: user_model)

}